db/sqlglue.c

```diff
@@ -5006,10 +4957,12 @@ int sqlite3BtreeCommit(Btree *pBt)
     if (clnt->selectv_arr)
         currangearr_coalesce(clnt->selectv_arr);
 
+    if (clnt->ctrl_sqlengine == SQLENG_NORMAL_PROCESS) {
```

Comment:
This is too early, we need first to commit, make sure we don't repeat if verify, than reset has_recording.
Example:
"insert into ... selectv ..." gets verify error